raster/src/main/scala/geotrellis/raster/io/geotiff/reader/TiffTagsReader.scala

```diff
@@ -106,7 +113,7 @@ object TiffTagsReader {
         byteBuffer.readDoublesTag(tiffTags, tagMetadata)
     }
 
-  implicit class ByteBufferTagReaderWrapper(val byteBuffer: ByteBuffer) extends AnyVal {
+  implicit class ByteReaderTagReaderWrapper(val byteBuffer: ByteReader) extends AnyVal {
```

Comment:
Extending `AnyVal` here would only affect the class as a structure, not what its methods could do. 

http://www.scala-lang.org/api/current/#scala.AnyVal

Sounds like it could be an optimization for runtime behaviour. @lossyrob ?